File: truevfs-kernel/truevfs-kernel-impl/src/main/scala/net/truevfs/kernel/impl/DefaultManagerFactory.scala
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.truevfs.kernel.impl

import javax.annotation.concurrent._
import net.truevfs.kernel.spec._
import net.truevfs.kernel.spec.spi._

/** Creates [[net.truevfs.kernel.impl.DefaultManager]] objects.
  * 
  * @author Christian Schlichtherle
  */
@Immutable
final class DefaultManagerFactory extends FsManagerFactory {
  override def apply = new DefaultManager
}

File: truevfs-kernel/truevfs-kernel-impl/src/main/java/net/java/truevfs/kernel/impl/DefaultManagerFactory.java
/*
 * Copyright © 2005 - 2021 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.kernel.impl;

import global.namespace.service.wight.annotation.ServiceImplementation;
import net.java.truevfs.kernel.spec.FsManager;
import net.java.truevfs.kernel.spec.spi.FsManagerFactory;

/**
 * Creates a default file system manager.
 *
 * @author Christian Schlichtherle
 */
@ServiceImplementation(priority = -100)
public final class DefaultManagerFactory implements FsManagerFactory {

    @Override
    public FsManager get() {
        return new DefaultManager();
    }
}
